QualityControl/test/testUtils/delay.js

```diff
@@ -17,3 +17,21 @@
  * @returns {Promise<void>}
  */
 export const delay = (ms = 500) => new Promise((resolve) => setTimeout(() => resolve(), ms));
+
+/**
+ * Delay until a condition is met or a timeout occurs
+ * @param {Function} conditionFunc - Function that returns a boolean indicating if the condition is met
+ * @param {number} ms - Time in milliseconds to wait between checks
+ * @param {number} maxRetries - Maximum number of retries before giving up
+ * @returns {Promise<void>}
+ */
+export const delayAndCheck = async (conditionFunc, ms = 500, maxRetries = 10) => {
+  let retries = 0;
+  while (retries < maxRetries) {
+    if (conditionFunc()) {
```

Comment:
if an async function is passed it will always be truthy... I would do:
```suggestion
    const result = await conditionFunc()
    if (result) {
```